pkg/kube/kclient/crdwatcher.go

```diff
@@ -104,7 +104,10 @@ func (c *crdWatcher) WaitForCRD(s schema.GroupVersionResource, stop <-chan struc
 func (c *crdWatcher) KnownOrCallback(s schema.GroupVersionResource, f func(stop <-chan struct{})) bool {
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
-	if c.HasSynced() && c.known(s) {
+	// We use HasStoreSyncedIgnoringHandlers here to avoid a subtle race condition.
+	// If we wait for the handlers to be called, we may have cloned the current callbacks in the handler, but not yet finished yet.
+	// In this case, we will add a callback, which will never be triggered later.
```

Comment:
This is referring to the callback on L116? And it wrote be triggered because of the callback slice clone on L55?